pkg/datapath/node.go

```diff
@@ -133,4 +133,10 @@ type NodeHandler interface {
 	// NodeConfigurationChanged is called when the local node configuration
 	// has changed
 	NodeConfigurationChanged(config LocalNodeConfiguration) error
+
+	// NodeNeighDiscoveryEnabled returns whether node neighbor discovery is enabled
+	NodeNeighDiscoveryEnabled() bool
```

Comment:
Hmm... This interface is becoming too broad leading to annoying no-op methods in the modules that implement this interface, but don't need some of the methods. I've made a note to see how we can refactor this code.